addons/netfox.extras/state-machine/network-state.gd

```diff
@@ -0,0 +1,33 @@
+@icon("res://addons/netfox/icons/state-synchronizer.svg")
+extends Node
+class_name NetworkedState
+
+@export var state_machine: NetworkedStateMachine
+
+## A networked state for use with [NetworkedStateMachine].
+
+func _ready():
+	if state_machine == null and get_parent() is NetworkedStateMachine:
+		state_machine = get_parent()
+
+# Custom update callback method
+func update(delta: float, tick: int, is_fresh: bool) -> void:
+	pass
+
+# Custom physics process callback method
+func physics_process(delta: float) -> void:
+	pass
+
+# Callback for entering the state
+func enter() -> void:
```

Comment:
What do you think about passing `tick` and `is_fresh` to both `enter()` and `exit()`? Also possibly the previous state? Not sure whether it would be beneficial for users, or just a bunch of ignored params.

Reply:
Right, `is_fresh` would be too much effort for too little benefit probably. We can skip that. You can grab tick from `NetworkRollback.tick`.